include/ruby/internal/intern/vm.h

```diff
@@ -253,15 +253,11 @@ void rb_undef_alloc_func(VALUE klass);
 rb_alloc_func_t rb_get_alloc_func(VALUE klass);
 
 /**
- * Clears  the constant  cache.   Extension libraries  should  not bother  such
- * things.   Just forget  about this  API (or  even, the  presence of  constant
- * cache).
- *
- * @internal
- *
- * Completely no idea why this function is defined in vm_method.c.
+ * Clears the inline constant caches associated with a particular ID. Extension
+ * libraries should not bother with such things. Just forget about this API (or
+ * even, the presence of constant caches).
  */
-void rb_clear_constant_cache(void);
+void rb_clear_constant_cache(ID id);
```

Comment:
Or even, let `rb_clear_constant_cache` be a no-op function and have `ID`-taking variant in a different name.

Reply:
If it's internal and we've removed all usage of this function, do we still need to mark it as deprecated?

Reply:
This declaration has been visible from extension libraries in the wild.  Deleting it could break them.  It is not a bad idea to tell them that we want to delete it, before actually doing so.

Reply:
I agree with @shyouhei that you should make the old one as deprecated and give the new one which takes an id parameter a different name.